lib/Doctrine/ORM/Internal/Hydration/ScalarColumnHydrator.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Internal\Hydration;
+
+use Doctrine\DBAL\Driver\Exception;
+use Doctrine\ORM\Exception\MultipleSelectorsFoundException;
+
+use function count;
+use function current;
+use function is_array;
+
+/**
+ * Hydrator that produces one-dimensional array.
+ */
+final class ScalarColumnHydrator extends AbstractHydrator
```

Comment:
I'm going to make a separate PR that should help with these static analysis issues.

Reply:
Thank you 🙏🏻

Reply:
Once https://github.com/doctrine/orm/pull/8945 is merged, it will be easy to fix the issues you have here

Reply:
Please rebase and use `statement()` and `resultSetMapping()` instead of `_stmt` and `_rsm`.

Reply:
rebased the branch, and use these getters